### PR TITLE
Add unit tests for org.wso2.lsp4intellij.utils.Utils

### PR DESCRIPTION
**Pull request by ThomasPerkins1123**
Hi,

I've analysed your code base and noticed that `org.wso2.lsp4intellij.utils.Utils` is not fully tested.

I've written some tests that cover this class with the help of [Diffblue Cover](https://www.diffblue.com/opensource).

Hopefully, these tests should help you detect any regressions caused by future code changes. If you would find it useful to have additional tests written for this repository, I would be more than happy to look at other particular classes that you consider important.

After inspection of this class I believe that your `arrayToString` method is probably not functioning as expected but is not used.

### CODE REVIEW

File: src/test/java/org/wso2/lsp4intellij/utils/UtilsTest.java [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+
+import org.junit.Assert;
+import org.junit.Test;
+import java.util.ArrayList;
+
+public class UtilsTest {
+
+  @Test
+  public void testConcatenateArrays() {
+    Assert.assertArrayEquals(new Object[]{new String[]{"a", "b"},
```
Comment (freddytuxworth) [inline src/test/java/org/wso2/lsp4intellij/utils/UtilsTest.java]: You can do `new String[][] {{"a", "b"}, {"c", "d"}}` instead of using nested `String[]` in `Object[]`

File: src/test/java/org/wso2/lsp4intellij/utils/UtilsTest.java [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+            new String[]{"c", "d"}}, new Utils().concatenateArrays(
+                    new String[]{"a", "b"}, new String[]{"c", "d"}));
+  }
+
+  @Test
+  public void testParseArgsSingleQuote() {
+    Assert.assertArrayEquals(new String[] {"\'\'"}, Utils.parseArgs(
+            new String[]{"\'\'"}));
```
Comment (Braavos6) [inline src/test/java/org/wso2/lsp4intellij/utils/UtilsTest.java]: Personal preference but I normally start a new line after the comma in the Assert.
`Assert.assertArrayEquals(new String[] {"\'\'"},`
`_________Utils.parseArgs(new String[]{"\'\'"}));`

File: src/test/java/org/wso2/lsp4intellij/utils/UtilsTest.java [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+
+  @Test
+  public void testParseArgsC() {
+    Assert.assertArrayEquals(new String[] {"c"}, Utils.parseArgs(
+            new String[]{"c"}));
+  }
+
+  @Test
+  public void testStringToList() {
+    // Assert result
```
Comment (Braavos6) [inline src/test/java/org/wso2/lsp4intellij/utils/UtilsTest.java]: you can delete this comment

**Review by freddytuxworth — CHANGES_REQUESTED** [review]

File: src/test/java/org/wso2/lsp4intellij/utils/UtilsTest.java [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+
+  @Test
+  public void testParseArgsSingleQuote() {
+    Assert.assertArrayEquals(new String[] {"\'\'"}, Utils.parseArgs(
+            new String[]{"\'\'"}));
+  }
+
+  @Test
+  public void testParseArgsDoubleQuote() {
+    Assert.assertArrayEquals(new String[] {"\"\""}, Utils.parseArgs(
```
Comment (Braavos6) [inline src/test/java/org/wso2/lsp4intellij/utils/UtilsTest.java]: I would also do some tests with null and/or empty strings :)

**Review by EricHetti — CHANGES_REQUESTED** [review]

File: src/test/java/org/wso2/lsp4intellij/utils/UtilsTest.java (hunk 4 above)
Comment (ThomasPerkins1123) [reply src/test/java/org/wso2/lsp4intellij/utils/UtilsTest.java]: There is no default in the case statement to handle empty strings so these would just be ignored and so i would argue that this is redundant.

**Review by paulkennethkent — APPROVED** [review]

File: src/test/java/org/wso2/lsp4intellij/utils/UtilsTest.java (hunk 4 above)
Comment (Braavos6) [reply src/test/java/org/wso2/lsp4intellij/utils/UtilsTest.java]: Right... I put that in the wrong line.. the methods `arrayToString()` and `stringToList() `have a null check :)

Also I am not sure if you are testing `arrayToString()` (I did a find on chrome and didn't find it mentioned)

Comment (ThomasPerkins1123) [reply src/test/java/org/wso2/lsp4intellij/utils/UtilsTest.java]: Ive added a null test for `stringToList()` but nothing is being done about `arrayToString()` as this method does not appear to work correctly and Freddy agrees.
I did just write a little not to the owner in the PR text comment to inform them of this

Comment (Braavos6) [reply src/test/java/org/wso2/lsp4intellij/utils/UtilsTest.java]: Alright :)

File: src/test/java/org/wso2/lsp4intellij/utils/UtilsTest.java [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+            "foo, Bar", ", "));
+  }
+
+  @Test
+  public void testStringToListSepNull() {
+    final List<String> arrayList = new ArrayList<>();
+    arrayList.add("fooBar");
+
+    Assert.assertEquals(arrayList, new Utils().stringToList(
+            "fooBar", null));
```
Comment (Braavos6) [inline src/test/java/org/wso2/lsp4intellij/utils/UtilsTest.java]: Change the alignment here so the line starts after the comma :)

File: src/test/java/org/wso2/lsp4intellij/utils/UtilsTest.java [hunk 6]
```diff
@@ -0,0 +1,65 @@
...
+  }
+
+  @Test
+  public void testStringToList() {
+    final List<String> arrayList = new ArrayList<>();
+    arrayList.add("foo");
+    arrayList.add("Bar");
+    
+    Assert.assertEquals(arrayList, new Utils().stringToList(
+            "foo, Bar", ", "));
```
Comment (Braavos6) [inline src/test/java/org/wso2/lsp4intellij/utils/UtilsTest.java]: Change the alignment here so the line starts after the comma

**Review by EricHetti — APPROVED** [review]

**Review by paulkennethkent — APPROVED** [review]

File: src/test/java/org/wso2/lsp4intellij/utils/UtilsTest.java [hunk 7]
```diff
@@ -0,0 +1,65 @@
...
+import org.junit.Assert;
+import org.junit.Test;
+import java.util.ArrayList;
+import java.util.List;
+
+public class UtilsTest {
+
+  @Test
+  public void testConcatenateArrays() {
+    Assert.assertArrayEquals(new Object[][]{{"a", "b"},
```
Comment (freddytuxworth) [inline src/test/java/org/wso2/lsp4intellij/utils/UtilsTest.java]: break line before `new Object[]`

**Review by freddytuxworth — CHANGES_REQUESTED** [review]

File: src/test/java/org/wso2/lsp4intellij/utils/UtilsTest.java [hunk 8]
```diff
@@ -0,0 +1,66 @@
+package org.wso2.lsp4intellij.utils;
```
Comment (Braavos6) [inline src/test/java/org/wso2/lsp4intellij/utils/UtilsTest.java:1]: Licence header file is missing

File: src/test/java/org/wso2/lsp4intellij/utils/UtilsTest.java [hunk 9]
```diff
@@ -0,0 +1,66 @@
...
+  @Test
+  public void testParseArgsSlash() {
+    Assert.assertArrayEquals(new String[] {"\\\\"},
+            Utils.parseArgs(new String[]{"\\\\"}));
+  }
+
+  @Test
+  public void testParseArgsC() {
+    Assert.assertArrayEquals(new String[] {"c"},
+            Utils.parseArgs(new String[]{"c"}));
```
Comment (freddytuxworth) [inline src/test/java/org/wso2/lsp4intellij/utils/UtilsTest.java:45]: be consistent with/without space between `[]` and `{`

File: src/test/java/org/wso2/lsp4intellij/utils/UtilsTest.java [hunk 10]
```diff
@@ -0,0 +1,66 @@
...
+  @Test
+  public void testParseArgsSpace() {
+    Assert.assertArrayEquals(new String[] {"", "\" \""},
+            Utils.parseArgs(new String[]{" \" \""}));
+  }
+
+  @Test
+  public void testParseArgsSlash() {
+    Assert.assertArrayEquals(new String[] {"\\\\"},
+            Utils.parseArgs(new String[]{"\\\\"}));
```
Comment (freddytuxworth) [inline src/test/java/org/wso2/lsp4intellij/utils/UtilsTest.java:39]: be consistent with/without space between `[]` and `{`

File: src/test/java/org/wso2/lsp4intellij/utils/UtilsTest.java [hunk 11]
```diff
@@ -0,0 +1,66 @@
...
+  @Test
+  public void testParseArgsDoubleQuote() {
+    Assert.assertArrayEquals(new String[] {"\"\""},
+            Utils.parseArgs(new String[]{"\"\""}));
+  }
+
+  @Test
+  public void testParseArgsSpace() {
+    Assert.assertArrayEquals(new String[] {"", "\" \""},
+            Utils.parseArgs(new String[]{" \" \""}));
```
Comment (freddytuxworth) [inline src/test/java/org/wso2/lsp4intellij/utils/UtilsTest.java:33]: be consistent with/without space between `[]` and `{`

File: src/test/java/org/wso2/lsp4intellij/utils/UtilsTest.java [hunk 12]
```diff
@@ -0,0 +1,66 @@
...
+  @Test
+  public void testParseArgsSingleQuote() {
+    Assert.assertArrayEquals(new String[] {"\'\'"},
+            Utils.parseArgs(new String[]{"\'\'"}));
+  }
+
+  @Test
+  public void testParseArgsDoubleQuote() {
+    Assert.assertArrayEquals(new String[] {"\"\""},
+            Utils.parseArgs(new String[]{"\"\""}));
```
Comment (freddytuxworth) [inline src/test/java/org/wso2/lsp4intellij/utils/UtilsTest.java:27]: be consistent with/without space between `[]` and `{`

File: src/test/java/org/wso2/lsp4intellij/utils/UtilsTest.java [hunk 13]
```diff
@@ -0,0 +1,66 @@
...
+    Assert.assertArrayEquals(
+            new Object[][]{{"a", "b"}, {"c", "d"}}, 
+            new Utils().concatenateArrays(
+                    new String[]{"a", "b"}, new String[]{"c", "d"}));
+  }
+
+  @Test
+  public void testParseArgsSingleQuote() {
+    Assert.assertArrayEquals(new String[] {"\'\'"},
+            Utils.parseArgs(new String[]{"\'\'"}));
```
Comment (freddytuxworth) [inline src/test/java/org/wso2/lsp4intellij/utils/UtilsTest.java:21]: be consistent with/without space between `[]` and `{`

**Review by freddytuxworth — CHANGES_REQUESTED** [review]

**Review by JohnLBergqvist — APPROVED** [review]

**Review by freddytuxworth — APPROVED** [review]
